util/types/time.go

```diff
@@ -1882,18 +1930,22 @@ func matchDateWithToken(t *mysqlTime, date string, token string) (remain string,
 }
 
 func parseTwoDigits(input string) (int, bool) {
-	if len(input) < 2 {
+	return parseSomeDigits(input, 2)
```

Comment:
it's used in many place, such as parsing hour, minute, minute in "00..xx" format...